docs/extensions/general-guidelines.md

```diff
@@ -0,0 +1,274 @@
+
+1.  Incorporate the feedback and strive for consensus.
+
+1.  Help to merge the extension into the main specification (this will require
+    converting the proposal to Markdown and submitting a Pull Request at
+    [the bids specification repository])
+
+1.  Create example datasets.
+
+1.  If necessary, contribute a pull request to the
+      [BIDS Validator repository](https://github.com/bids-standard/bids-validator)
+      as well to incorporate the extension.
+
+![bep_process](../assets/img/bep_process.png)
+
+## Advice for extending BIDS
+
+### Limit flexibility, consider tool developers
+
+One of the aims of BIDS is to make reusing data easier.
+This means that when proposing an extension you need to put yourself
+in the shoes of someone who will receive a BIDS dataset and attempt to analyze it.
+Additionally, consider developers that will try to write tools that take BIDS datasets as inputs.
+It is worth assessing how much additional code different ways of approaching your extension may cause.
+
+The most common situation where the trade-off between flexibility and ease of
+tool building comes up is in choosing file formats.
+For example, allowing multiple different file formats to be used to represent the same data type is flexible,
+but requires developers to provide support for all of them.
+As an example, iEEG-BIDS and EEG-BIDS
+surveyed the community <!--TODO find links  -->
+to find out about most common formats and selected only a few formats based on usage and their openness.
+
+### Get the community involved
+
+Try to reach out to colleagues working with the type of data you are trying to add support for.
+The more people looking at your extension the better it will become through discussions.
+
+### Be consistent with the main specification
+
+The main specification follows some general rules.
+For example, see the [rules on participant labels](https://bids-specification.readthedocs.io/en/stable/02-common-principles.html#participant-names-and-other-labels).
+
+Try not to deviate from BIDS conventions in your extension.
+
+### Be consistent with other BEPs
+
+A common dictionary (BIDS keys) is what makes BIDS successful,
+it is thus essential to not create many new entities.
+Many of the current BEPs have developed useful terms that we recommend here:
+
+| Entity                                               | BEP(s)                                                                                   | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
```

Comment:
divide this table into entities that are already merged and those that are not

possibly generate the first from from the schema and point to the relevant pages in the bids spec